dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-prometheus/src/main/java/org/apache/dolphinscheduler/plugin/alert/prometheus/AlertManagerSender.java

```diff
@@ -0,0 +1,168 @@
+                        alertData.getContent());
+                return resp;
+            }
+
+            log.error("send prometheus alert manager message error, return http status code: {} ", statusCode);
+            try {
+                HttpEntity entity = response.getEntity();
+                resp = EntityUtils.toString(entity, "utf-8");
+                EntityUtils.consume(entity);
+            } finally {
+                response.close();
+            }
+            log.info("Prometheus alert manager send alert failed, title: {} ,content: {}, resp: {}",
+                    alertData.getTitle(),
+                    alertData.getContent(), resp);
+
+            return resp;
+        } finally {
+            httpClient.close();
+        }
+    }
+
+    public AlertResult checkSendAlertManageMsgResult(String resp) {
+        AlertResult alertResult = new AlertResult();
+        alertResult.setStatus("false");
+
+        if (Objects.equals(resp, AlertManagerConstants.ALERT_SUCCESS)) {
+            alertResult.setStatus("true");
+            alertResult.setMessage("prometheus alert manager send success");
+            return alertResult;
+        }
+
+        alertResult.setMessage("prometheus alert manager send fail");
+        log.info("send prometheus alert manager msg error,resp error");
+        return alertResult;
+    }
+
+    public String generateContentJson(AlertData alertData) {
+        List<HashMap> list = JSONUtils.toList(alertData.getContent(), HashMap.class);
+        HashMap<String, String> labels = new HashMap<>();
+        if (CollectionUtils.isEmpty(list)) {
+            labels.put("content", alertData.getContent());
+        }
+        for (Map map : list) {
+            for (Map.Entry<String, Object> entry : (Iterable<Map.Entry<String, Object>>) map.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue().toString();
+                labels.put(key, value);
+            }
+        }
+        labels.put("title", alertData.getTitle());
+
+        Map<String, Object> alert = new HashMap<>();
+        alert.put("labels", labels);
+
+        Map<String, String> annotations = JSONUtils.toMap(this.annotations);
+        if (annotations != null) {
+            alert.put("annotations", annotations);
+        }
+
+        String formattedTime = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'").format(new Date());
+        alert.put("startsAt", formattedTime);
+        alert.put("endsAt", formattedTime);
+
+        if (generatorURL != null && !generatorURL.equals("")) {
+            alert.put("generatorURL", generatorURL);
+        }
+        List<Map<String, Object>> body = new ArrayList<>();
+        body.add(alert);
+        return JSONUtils.toJsonString(body);
+    }
+
+    private static CloseableHttpClient getDefaultClient() {
+        return HttpClients.createDefault();
```

Comment:
https://github.com/apache/dolphinscheduler/blob/a5284e40242eea7597e7be27d62e3133546e7073/dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java#L110-L116

Based on the implementation of other plugins, it is recommended to use an HTTP client that supports retrying